Toxicity Measurement (#262)
huggingface/evaluate: measurements/toxicity/README.md

## How to use

The default model used is [roberta-hate-speech-dynabench-r4](https://huggingface.co/facebook/roberta-hate-speech-dynabench-r4-target). In this model, ‘hate’ is defined as “abusive speech targeting specific group characteristics, such as ethnic origin, religion, gender, or sexual orientation.” Definitions used by other classifiers may vary.
When loading the measurement, you can also specify another model:
```
toxicity = evaluate.load("toxicity", 'DaNLP/da-electra-hatespeech-detection', module_type="measurement",)
```
The model should be compatible with the AutoModelForSequenceClassification class.
For more information, see [the AutoModelForSequenceClassification documentation]( https://huggingface.co/docs/transformers/master/en/model_doc/auto#transformers.AutoModelForSequenceClassification).

Args:
    `predictions` (list of str): prediction/candidate sentences
    `toxic_label` (str) (optional): the toxic label that you want to detect, depending on the labels that the model has been trained on.
        This can be found using the `id2label` function, e.g.:

>         ```

Reviewer: ```suggestion
        ```python
```

        >>> model.config.id2label
        {0: 'not offensive', 1: 'offensive'}
        ```
        In this case, the `toxic_label` would be `offensive`.
    `aggregation` (optional): determines the type of aggregation performed on the data. If set to `None`, the scores for each prediction are returned.
    Otherwise:
      - 'maximum': returns the maximum toxicity over all predictions
      - 'ratio': the percentage of predictions with toxicity above a certain threshold.
    `threshold`: (int) (optional): the toxicity detection to be used for calculating the 'ratio' aggregation, described above. The default threshold is 0.5, based on the one established by [RealToxicityPrompts](https://arxiv.org/abs/2009.11462).

## Output values

    `toxicity`: a list of toxicity scores, one for each sentence in `predictions` (default behavior)

    `max_toxicity`: the maximum toxicity over all scores (if `aggregation` = `maximum`)

    `toxicity_ratio` : the percentage of predictions with toxicity >= 0.5 (if `aggregation` = `ratio`)

### Values from popular papers

## Examples
    Example 1 (default behavior):

> ```

Reviewer: ```suggestion
```python
```

>>> toxicity = evaluate.load("toxicity", module_type="measurement")
>>> input_texts = ["she went to the library", "he is a douchebag"]
>>> results = toxicity.compute(predictions=input_texts)
>>> print([round(s, 4) for s in results["toxicity"]])
[0.0002, 0.8564]
```
    Example 2 (returns ratio of toxic sentences):

> ```

Reviewer: ```suggestion
```python
```

>>> toxicity = evaluate.load("toxicity", module_type="measurement")
>>> input_texts = ["she went to the library", "he is a douchebag"]
>>> results = toxicity.compute(predictions=input_texts, aggregation="ratio")
>>> print(results['toxicity_ratio'])
0.5
```
    Example 3 (returns the maximum toxicity score):

> ```

Reviewer: ```suggestion
```python
```

>>> toxicity = evaluate.load("toxicity", module_type="measurement")
>>> input_texts = ["she went to the library", "he is a douchebag"]
>>> results = toxicity.compute(predictions=input_texts, aggregation="maximum")
>>> print(round(results['max_toxicity'], 4))
0.8564
```
    Example 4 (uses a custom model):

> ```

Reviewer: ```suggestion
```python
```

---
title: Toxicity
emoji: 🤗
colorFrom: blue
colorTo: red
sdk: gradio
sdk_version: 3.0.2
app_file: app.py
pinned: false
tags:
- evaluate
- measurement
description: >-

> The toxicity measurement aims to quantify the toxicity of the input texts using a hate speech classification model trained for the task

Reviewer 2: ```suggestion
The toxicity measurement aims to quantify the toxicity of the input texts using a pretrained hate speech classification model.
```

            >>> model.config.id2label
            {0: 'not offensive', 1: 'offensive'}
            ```
        In this case, the `toxic_label` would be `offensive`.
    `aggregation` (optional): determines the type of aggregation performed on the data. If set to `None`, the scores for each prediction are returned.
     Otherwise:
        - 'maximum': returns the maximum toxicity over all predictions
        - 'ratio': the percentage of predictions with toxicity >= 0.5.

## Output values

    `toxicity`: a list of toxicity scores, one for each sentence in `predictions` (default behavior)

    `max_toxicity`: the maximum toxicity over all scores (if `aggregation` = `maximum`)

    `toxicity_ratio` : the percentage of predictions with toxicity >= 0.5 (if `aggregation` = `ratio`)

### Values from popular papers

## Examples
    Example 1 (default behavior):
```
>>> toxicity = evaluate.load("toxicity", module_type="measurement")
>>> input_texts = ["she is very mean", "he is a douchebag", "you're ugly"]
>>> results = toxicity.compute(predictions=input_texts)
>>> print(results)

> {'toxicity': [0.00013419731112662703, 0.856372594833374, 0.0020856475457549095]}

Author: changed!
